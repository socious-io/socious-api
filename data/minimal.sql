CREATE TABLE public.users(
  id serial PRIMARY KEY,
  first_name varchar(70) DEFAULT NULL,
  username varchar(200) UNIQUE NOT NULL,
  name varchar(255) DEFAULT NULL,
  email varchar(200) UNIQUE NOT NULL,
  email_text varchar(255) DEFAULT NULL,
  mobile_countries_id int(11) DEFAULT NULL,
  phone varchar(255) DEFAULT NULL,
  wallet_address varchar(255) DEFAULT NULL,
  password text NOT NULL,
  remember_token text DEFAULT NULL,
  city text,
  city_ja mediumtext,
  description_search mediumtext,
  description_search_ja mediumtext,
  address_detail text,
  avatar text,
  cover_image text,
  expiry_date datetime DEFAULT NULL,
  active boo,
  activation_token varchar(255) DEFAULT NULL,
  mission text,
  bio mediumtext,
  profile_id varchar(200) DEFAULT NULL,
  view_as int(11) DEFAULT NULL,
  language varchar(255) DEFAULT NULL,
  my_conversation varchar(255) DEFAULT NULL,  
  email_verified_at datetime DEFAULT NULL,
  impact_score float NOT NULL DEFAULT '0',
  lock_otp datetime DEFAULT NULL,
  created_at datetime NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
  updated_at datetime NOT NULL DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(6),
  deleted_at datetime DEFAULT NULL
);
