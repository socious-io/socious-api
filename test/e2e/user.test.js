import sql from 'sql-template-tag'
import supertest from 'supertest'
import { app } from '../../src/index.js'
import data from '../data/index.js'
import {
  login,
  register,
  profile,
  addLanguage,
  addExperience,
  updateExperience,
  updateLanguage,
  profileByUsername,
  updateProfile,
  verifyUser,
  updateUserWallet,
  openToWork,
  openToVolunteer,
  registerReferredBy,
  requestExperienceCredentials,
  approveRequestedExperienceCredentials
} from './globals/user.js'

let server, request

beforeAll(async () => {
  server = app.listen()
  request = supertest(server)
})

test('register', async () => register(request, data))
test('verify', async () => verifyUser(request, data))
test('login', async () => login(request, data))
test('add language', async () => addLanguage(request, data))
test('add experience', async () => addExperience(request, data))
test('profile', async () => profile(request, data))
test('update language', async () => updateLanguage(request, data))
test('update experience', async () => updateExperience(request, data))
test('request experience credentials', async () => requestExperienceCredentials(request, data))
test('approve request experience credentials', async () => approveRequestedExperienceCredentials(request, data))
test('profile by username', async () => profileByUsername(request, data))
test('update profile', async () => updateProfile(request, data))
test('update wallet', async () => updateUserWallet(request, data))
test('open to work', async () => openToWork(request, data))
test('open to volunteer', async () => openToVolunteer(request, data))
test('register referred by another user', async () => registerReferredBy(request, data))

test('delete user', async () => {
  const response = await request
    .post('/user/delete')
    .send({
      reason: 'TEST'
    })
    .set('Authorization', data.users[0].access_token)

  expect(response.status).toBe(200)

  const deletedUser = await app.db.get(sql`SELECT * FROM deleted_users WHERE username=${data.users[0].username}`)

  expect(deletedUser).toMatchSnapshot({
    id: expect.any(String),
    user_id: expect.any(String),
    registered_at: expect.any(Object),
    deleted_at: expect.any(Object)
  })
})

const cleanup = async () => {
  await app.db.query('DELETE FROM users')
  await app.db.query('DELETE FROM organizations')
  await app.db.pool.end()
}

afterAll((done) => {
  cleanup(done).then(() => {
    server.close(done)
  })
})
