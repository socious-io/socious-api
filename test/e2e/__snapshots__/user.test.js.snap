// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add experience 1`] = `
{
  "created_at": Any<String>,
  "description": "Test",
  "end_at": null,
  "id": Any<String>,
  "org_id": Any<String>,
  "skills": null,
  "start_at": Any<String>,
  "title": "Test",
  "user_id": Any<String>,
}
`;

exports[`add experience 2`] = `
{
  "created_at": Any<String>,
  "description": "Test",
  "end_at": null,
  "id": Any<String>,
  "org_id": Any<String>,
  "skills": null,
  "start_at": Any<String>,
  "title": "Test",
  "user_id": Any<String>,
}
`;

exports[`add language 1`] = `
{
  "created_at": Any<String>,
  "id": Any<String>,
  "level": "NATIVE",
  "name": "FA",
  "user_id": Any<String>,
}
`;

exports[`add language 2`] = `
{
  "created_at": Any<String>,
  "id": Any<String>,
  "level": "NATIVE",
  "name": "FA",
  "user_id": Any<String>,
}
`;

exports[`delete user 1`] = `
{
  "deleted_at": Any<Object>,
  "id": Any<String>,
  "reason": "TEST",
  "registered_at": Any<Object>,
  "user_id": Any<String>,
  "username": "test_username",
}
`;

exports[`login 1`] = `
{
  "error": ""email" must be a valid email",
}
`;

exports[`profile 1`] = `
{
  "address": null,
  "avatar": null,
  "bio": null,
  "city": null,
  "country": null,
  "cover_image": null,
  "created_at": Any<String>,
  "experiences": [
    {
      "description": "Test",
      "end_at": null,
      "id": Any<String>,
      "org": {
        "address": "address",
        "bio": "test",
        "city": "Tokyo",
        "country": "JP",
        "cover_image": null,
        "created_at": Any<String>,
        "created_by": Any<String>,
        "culture": "test",
        "description": "test",
        "email": "test@socious.io",
        "followers": 1,
        "followings": 0,
        "geoname_id": null,
        "hiring": false,
        "id": Any<String>,
        "image": null,
        "impact_points": 0,
        "mission": "test",
        "mobile_country_code": null,
        "name": "test ORG",
        "other_party_id": null,
        "other_party_title": null,
        "other_party_url": null,
        "phone": "9379911",
        "search_tsv": Any<String>,
        "shortname": Any<String>,
        "social_causes": [
          "SOCIAL",
          "HEALTH",
        ],
        "status": "ACTIVE",
        "type": "COOP",
        "updated_at": Any<String>,
        "verified_impact": false,
        "wallet_address": null,
        "website": "http://test.com",
      },
      "skills": null,
      "start_at": "2022-10-16T13:32:30.211",
      "title": "Test",
    },
  ],
  "first_name": "test_name",
  "followers": 0,
  "followings": 2,
  "geoname_id": null,
  "id": Any<String>,
  "impact_points": 0,
  "languages": [
    {
      "id": Any<String>,
      "level": "NATIVE",
      "name": "FA",
    },
  ],
  "last_name": "last_name",
  "mission": null,
  "mobile_country_code": null,
  "open_to_volunteer": false,
  "open_to_work": false,
  "phone": null,
  "proofspace_connect_id": null,
  "reported": false,
  "skills": null,
  "social_causes": null,
  "username": "test_username",
  "wallet_address": null,
}
`;

exports[`profile 2`] = `
{
  "address": null,
  "avatar": null,
  "bio": null,
  "city": null,
  "country": null,
  "cover_image": null,
  "created_at": Any<String>,
  "experiences": [
    {
      "description": "Test",
      "end_at": null,
      "id": Any<String>,
      "org": {
        "address": "address",
        "bio": "test",
        "city": "Tokyo",
        "country": "JP",
        "cover_image": null,
        "created_at": Any<String>,
        "created_by": Any<String>,
        "culture": "test",
        "description": "test",
        "email": "test@socious.io",
        "followers": 1,
        "followings": 0,
        "geoname_id": null,
        "hiring": false,
        "id": Any<String>,
        "image": null,
        "impact_points": 0,
        "mission": "test",
        "mobile_country_code": null,
        "name": "test ORG",
        "other_party_id": null,
        "other_party_title": null,
        "other_party_url": null,
        "phone": "9379911",
        "search_tsv": Any<String>,
        "shortname": Any<String>,
        "social_causes": [
          "SOCIAL",
          "HEALTH",
        ],
        "status": "ACTIVE",
        "type": "COOP",
        "updated_at": Any<String>,
        "verified_impact": false,
        "wallet_address": null,
        "website": "http://test.com",
      },
      "skills": null,
      "start_at": "2022-10-16T13:32:30.211",
      "title": "Test",
    },
  ],
  "first_name": "test_name_2",
  "followers": 0,
  "followings": 0,
  "geoname_id": null,
  "id": Any<String>,
  "impact_points": 0,
  "languages": [
    {
      "id": Any<String>,
      "level": "NATIVE",
      "name": "FA",
    },
  ],
  "last_name": "last_name_2",
  "mission": null,
  "mobile_country_code": null,
  "open_to_volunteer": false,
  "open_to_work": false,
  "phone": null,
  "proofspace_connect_id": null,
  "reported": false,
  "skills": null,
  "social_causes": null,
  "username": "test_username_2",
  "wallet_address": null,
}
`;

exports[`profile by username 1`] = `
{
  "address": null,
  "avatar": null,
  "bio": null,
  "city": null,
  "country": null,
  "cover_image": null,
  "created_at": Any<String>,
  "experiences": [
    {
      "description": "Test",
      "end_at": null,
      "id": Any<String>,
      "org": {
        "address": "address",
        "bio": "test",
        "city": "Tokyo",
        "country": "JP",
        "cover_image": null,
        "created_at": Any<String>,
        "created_by": Any<String>,
        "culture": "test",
        "description": "test",
        "email": "test@socious.io",
        "followers": 1,
        "followings": 0,
        "geoname_id": null,
        "hiring": false,
        "id": Any<String>,
        "image": null,
        "impact_points": 0,
        "mission": "test",
        "mobile_country_code": null,
        "name": "test ORG",
        "other_party_id": null,
        "other_party_title": null,
        "other_party_url": null,
        "phone": "9379911",
        "search_tsv": Any<String>,
        "shortname": Any<String>,
        "social_causes": [
          "SOCIAL",
          "HEALTH",
        ],
        "status": "ACTIVE",
        "type": "COOP",
        "updated_at": Any<String>,
        "verified_impact": false,
        "wallet_address": null,
        "website": "http://test.com",
      },
      "skills": null,
      "start_at": Any<String>,
      "title": "Test",
    },
  ],
  "first_name": "test_name",
  "followers": 0,
  "followings": 2,
  "geoname_id": null,
  "id": Any<String>,
  "impact_points": 0,
  "languages": [
    {
      "id": Any<String>,
      "level": "PROFICIENT",
      "name": "EN",
    },
  ],
  "last_name": "last_name",
  "mission": null,
  "mobile_country_code": null,
  "open_to_volunteer": false,
  "open_to_work": false,
  "phone": null,
  "proofspace_connect_id": null,
  "reported": false,
  "skills": null,
  "social_causes": null,
  "username": "test_username",
  "wallet_address": null,
}
`;

exports[`profile by username 2`] = `
{
  "address": null,
  "avatar": null,
  "bio": null,
  "city": null,
  "country": null,
  "cover_image": null,
  "created_at": Any<String>,
  "experiences": [
    {
      "description": "Test",
      "end_at": null,
      "id": Any<String>,
      "org": {
        "address": "address",
        "bio": "test",
        "city": "Tokyo",
        "country": "JP",
        "cover_image": null,
        "created_at": Any<String>,
        "created_by": Any<String>,
        "culture": "test",
        "description": "test",
        "email": "test@socious.io",
        "followers": 1,
        "followings": 0,
        "geoname_id": null,
        "hiring": false,
        "id": Any<String>,
        "image": null,
        "impact_points": 0,
        "mission": "test",
        "mobile_country_code": null,
        "name": "test ORG",
        "other_party_id": null,
        "other_party_title": null,
        "other_party_url": null,
        "phone": "9379911",
        "search_tsv": Any<String>,
        "shortname": Any<String>,
        "social_causes": [
          "SOCIAL",
          "HEALTH",
        ],
        "status": "ACTIVE",
        "type": "COOP",
        "updated_at": Any<String>,
        "verified_impact": false,
        "wallet_address": null,
        "website": "http://test.com",
      },
      "skills": null,
      "start_at": Any<String>,
      "title": "Test",
    },
  ],
  "first_name": "test_name_2",
  "followers": 0,
  "followings": 0,
  "geoname_id": null,
  "id": Any<String>,
  "impact_points": 0,
  "languages": [
    {
      "id": Any<String>,
      "level": "PROFICIENT",
      "name": "EN",
    },
  ],
  "last_name": "last_name_2",
  "mission": null,
  "mobile_country_code": null,
  "open_to_volunteer": false,
  "open_to_work": false,
  "phone": null,
  "proofspace_connect_id": null,
  "reported": false,
  "skills": null,
  "social_causes": null,
  "username": "test_username_2",
  "wallet_address": null,
}
`;

exports[`register 1`] = `
{
  "error": ""email" must be a valid email",
}
`;

exports[`update experience 1`] = `
{
  "created_at": Any<String>,
  "description": "Test",
  "end_at": null,
  "id": Any<String>,
  "org_id": Any<String>,
  "skills": null,
  "start_at": Any<String>,
  "title": "Test",
  "user_id": Any<String>,
}
`;

exports[`update experience 2`] = `
{
  "created_at": Any<String>,
  "description": "Test",
  "end_at": null,
  "id": Any<String>,
  "org_id": Any<String>,
  "skills": null,
  "start_at": Any<String>,
  "title": "Test",
  "user_id": Any<String>,
}
`;

exports[`update language 1`] = `
{
  "created_at": Any<String>,
  "id": Any<String>,
  "level": "PROFICIENT",
  "name": "EN",
  "user_id": Any<String>,
}
`;

exports[`update language 2`] = `
{
  "created_at": Any<String>,
  "id": Any<String>,
  "level": "PROFICIENT",
  "name": "EN",
  "user_id": Any<String>,
}
`;

exports[`update profile 1`] = `
{
  "address": null,
  "avatar": null,
  "bio": "TEST",
  "city": null,
  "country": null,
  "cover_image": null,
  "created_at": Any<String>,
  "experiences": [
    {
      "description": "Test",
      "end_at": null,
      "id": Any<String>,
      "org": {
        "address": "address",
        "bio": "test",
        "city": "Tokyo",
        "country": "JP",
        "cover_image": null,
        "created_at": Any<String>,
        "created_by": Any<String>,
        "culture": "test",
        "description": "test",
        "email": "test@socious.io",
        "followers": 1,
        "followings": Any<Number>,
        "geoname_id": null,
        "hiring": false,
        "id": Any<String>,
        "image": null,
        "impact_points": 0,
        "mission": "test",
        "mobile_country_code": null,
        "name": "test ORG",
        "other_party_id": null,
        "other_party_title": null,
        "other_party_url": null,
        "phone": "9379911",
        "search_tsv": Any<String>,
        "shortname": Any<String>,
        "social_causes": [
          "SOCIAL",
          "HEALTH",
        ],
        "status": "ACTIVE",
        "type": "COOP",
        "updated_at": Any<String>,
        "verified_impact": false,
        "wallet_address": null,
        "website": "http://test.com",
      },
      "skills": null,
      "start_at": "2022-10-16T13:32:30.211",
      "title": "Test",
    },
  ],
  "first_name": "test_name",
  "followers": 0,
  "followings": 2,
  "geoname_id": null,
  "id": Any<String>,
  "impact_points": 0,
  "languages": [
    {
      "id": Any<String>,
      "level": "PROFICIENT",
      "name": "EN",
    },
  ],
  "last_name": "last_name",
  "mission": "TEST",
  "mobile_country_code": "+98",
  "open_to_volunteer": false,
  "open_to_work": false,
  "phone": "9999999999",
  "proofspace_connect_id": null,
  "reported": false,
  "skills": [],
  "social_causes": null,
  "username": "test_username",
  "wallet_address": null,
}
`;

exports[`update profile 2`] = `
{
  "error": "duplicate key value violates unique constraint "idx_phone_country_code"",
}
`;
